engineio/src/header.rs

```diff
@@ -0,0 +1,150 @@
+use crate::Error;
+use bytes::Bytes;
+use reqwest::header::{
+    HeaderMap as ReqwestHeaderMap, HeaderName as ReqwestHeaderName,
+    HeaderValue as ReqwestHeaderValue,
+};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::str::FromStr;
+use websocket::header::Headers as WebSocketHeaderMap;
+
+#[derive(Eq, PartialEq, Hash, Debug, Clone)]
+pub struct HeaderName {
+    inner: String,
+}
```

Comment:
How much would a usage of `&'a str` blow this up? Is it possible or just unnecessary because it produces a lot of `<'_>` bloat?

Reply:
Possibly a fairly large blow up and/or breakage as header names could have varying lifetimes, while the header map requires all the lifetimes of it's keys to be the same. Making a reference as a key to a map could lead to other strange behavior (like double references)

Reply:
Ah I see, makes sense. Then nevermind ;)